### ISSUE TITLE
7c31 A THR 44 CG2 hydrogens not rotated out of the way in reduce2

### ISSUE THREAD

**Issue by russell-taylor** [ReliaSolve/cctbx_project#232]
In reduce, these are rotated to not collide with A GLU 4 HG3, but in reduce2 they are left in the original staggered orientation.

A MoverTetrahedralMethylRotator was used to stagger the atoms in reduce2 but a Mover was added in reduce.

Asking Jane about this.

**Comment by russell-taylor** [comment]
This is expected behavior in Reduce2 as it usually indicates a problem with the carbon placement so should produce a clash.

**Comment by russell-taylor** [comment]
**@todo:** Actually, this is not expected behavior. These are not tetrahedral Methyls. There must be another check before placement of an aromatic Methyl rotator to see if the system is part of an Aromatic ring. If so, then we place an aromatic methyl rotator as is happening now. If not, then we place a general rotator.

**Comment by russell-taylor** [comment]
The dt residue found in 404d is an example of an aromatic ring methyl.

The ones on HEC in the fe6k snippet are also treated this way in original Reduce.

**Comment by russell-taylor** [comment]
Reduce code: If the bonded neighbor of the Methyl group has 4 bonded neighbors, then a rotator should not be added. **@todo:** This means that there should be no rotator on the A THR 44 CH2 in Reduce -- it should be staggered. Why is it rotated? There is not a Mover placed on this atom in Reduce (only a single rotator on OG1).

If it has 3 bonded neighbors (one will be a double bond if this neighbor is a  carbon) then we see if the carbon in the methyl is marked aromatic and if so we place an aromatic Methyl; **@todo:** otherwise, we place a general rotator.

**Comment by russell-taylor** [comment]
(done) The problem is that the stagger is not being performed correctly by Reduce in one of the files in C:\tmp. The angle is not correct. This was an older run, so maybe it has been fixed by now. Or maybe it is a bug in the current master of Reduce. It is a bug in the current master of Reduce.

When we run the regression test on the current master vs. a version from before I started changing things, on Linux, we get no differences in THR in that PDB file... this is when we pull from the reliasolve repository, with the last commit on Jul 12, 2022. When we check out that version on Windows, we still get improper staggering. This is also true in the new PDB generated from Linux and in the old one -- we need to go further back to find the change?

MolProbity ran with reduce.3.7.201124, so going to commit ce6a22d5721606ef04743d9378dc6cd7c6b59f43.  **It looks like the old version is not replacing all of the hydrogens -- they may have been there in the original file.**  They are in the original file! It looks like MolProbity is running -TRIM before running -FLIP!

**Solution:** Add `-DROP_HYDROGENS_ON_ATOM_RECORDS -DROP_HYDROGENS_ON_OTHER_RECORDS` to the command line on the master-branch Reduce.
